### PR TITLE
VIDEO-8282 - Adding iPad Detection

### PR DESCRIPTION
**Pull request by PikaJoyce**
JIRA Ticket : [VIDEO-8282](https://issues.corp.twilio.com/browse/VIDEO-8282)

Adding in iPad detection function to send to insights.

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Description

A description of what this PR does.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### CODE REVIEW

File: lib/util/index.js
Comment (charliesantos) [inline lib/util/index.js:141]: We should probably put this inside browserdetection.js

File: lib/util/insightspublisher/index.js
Comment (charliesantos) [inline lib/util/insightspublisher/index.js:66]: Why create another internal property? Can we call util.isIpad() directly on line 205?

File: lib/util/insightspublisher/index.js
Comment (charliesantos) [inline lib/util/insightspublisher/index.js:207]: Don't forget to remove commented out code.

File: test/unit/spec/util/index.js
Comment (charliesantos) [inline test/unit/spec/util/index.js:342]: We should move this in browserdetection file.

File: lib/util/insightspublisher/index.js
Comment (charliesantos) [inline lib/util/insightspublisher/index.js:205]: Please add unit tests for this.

**Review by charliesantos — CHANGES_REQUESTED** [review]

File: lib/util/index.js
Comment (makarandp0) [inline lib/util/index.js:144]: how about iPhone - do want to report iPhones as well?

File: lib/util/browserdetection.js
Comment (makarandp0) [inline lib/util/browserdetection.js:38]: just curious - does the width stay at `window.screen.width <=  476` when iPhone is rotated? same question for iPad

Comment (PikaJoyce) [reply lib/util/browserdetection.js:38]: Yes. Just tested this on my iPhone and it's the same value!

Comment (PikaJoyce) [reply lib/util/browserdetection.js:38]: I used the largest width from iPhones (iPhone 7 Plus was the last of this size!) and the smallest width from iPads (Starting from the "smallest" iPads which were 768 in width). 

However, I can go one bit smaller and that this helped me realize that iPad minis (744 width) which are no longer in production have a slightly smaller screen size but I'll include it as part of this.

Otherwise, there doesn't seem to be any devices currently available from Apple that have overlaps in widths for these devices.

File: test/unit/spec/util/index.js
Comment (PikaJoyce) [reply test/unit/spec/util/index.js:18]: Unsure how that even got added. Will remove this! Thanks!

File: lib/util/browserdetection.js
Comment (charliesantos) [inline lib/util/browserdetection.js:24]: `hasTouchScreen`?

File: lib/util/insightspublisher/index.js
Comment (charliesantos) [inline lib/util/insightspublisher/index.js:73]: @PikaJoyce I think you missed my comments here in my last 2 reviews :)
These internal properties are not needed right? Can we call isIpad() and isIphone() directly on line 211?

**Review by charliesantos — CHANGES_REQUESTED** [review]

File: lib/util/insightspublisher/index.js
Comment (PikaJoyce) [reply lib/util/insightspublisher/index.js:73]: I've added these back in order to unit test insightspublisher, so mostly for dependency injection.

**Review by charliesantos — APPROVED** [review]
Good job Joyce. Let's make sure tests passes and that we also get a +1 from @makarandp0 before merging.

File: lib/util/insightspublisher/index.js
Comment (makarandp0) [inline lib/util/insightspublisher/index.js:69]: @PikaJoyce , For testing, instead of passsing isIpad, isIphone in options, you can mock them using `sinon.stub` 
This lets you stub the whole `../browserdetection` module from the stub. That way you wont have to add these extra member in the class.  You can see an example of this here: https://github.com/twilio/twilio-video.js/blob/76dd81b6fcef180dbf9c030436e8a20a09639ac4/test/unit/spec/signaling/v2/peerconnection.js#L87

**Review by makarandp0 — COMMENTED** [review]
I have one suggestion for avoiding the _isIphone, _isIpad properties. but LGTM otherwise.

File: lib/util/insightspublisher/index.js
Comment (PikaJoyce) [reply lib/util/insightspublisher/index.js:69]: I'll try this out!

**Review by makarandp0 — APPROVED** [review]

**Pull request merged**
